snapcraft/internal/sources/__init__.py

```diff
@@ -186,8 +187,8 @@ def _get_source_type_from_uri(source, ignore_errors=False):  # noqa: C901
     elif source.endswith('7z'):
         source_type = '7z'
     elif common.isurl(source) and not ignore_errors:
-        raise ValueError('no handler to manage source ({})'.format(source))
+        raise errors.SnapcraftSourceUnhandledError(source)
     elif not os.path.isdir(source) and not ignore_errors:
-        raise ValueError('local source ({}) is not a directory'.format(source))
+        raise errors.SnapcraftSourceNotADirectoryError(source)
```

Comment:
this is a nit, but I would prefer SnapcraftSourceNotDictionaryError. Without the `a`.

Reply:
Given that python actually has a `NotADirectoryError` I personally think this one makes sense.